example/lib/edge.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_aepedge/flutter_aepedge_data.dart';
+import 'package:flutter_aepedge/flutter_aepedge.dart';
+import 'util.dart';
+
+class EdgePage extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<EdgePage> {
+  String _edgeVersion = 'Unknown';
+
+  List<EventHandle> _edgeEventHandleResponse = List.empty();
+  @override
+  void initState() {
+    super.initState();
+    initPlatformState();
+  }
+
+  // Platform messages are asynchronous, so we initialize in an async method.
+  Future<void> initPlatformState() async {
+    late String edgeVersion;
+    // Platform messages may fail, so we use a try/catch PlatformException.
+    try {
+      edgeVersion = await Edge.extensionVersion;
+    } on PlatformException {
+      log("Failed to get extension versions");
+    }
+
+    // If the widget was removed from the tree while the asynchronous platform
+    // message was in flight, we want to discard the reply rather than calling
+    // setState to update our non-existent appearance.
+    if (!mounted) return;
+
+    setState(() {
+      _edgeVersion = edgeVersion;
+    });
+  }
+
+  Future<void> sendEvent([datasetId]) async {
+    late List<EventHandle> result;
+    Map<dynamic, dynamic> xdmData = {"eventType": "SampleEventType"};
+    Map<String, dynamic> data = {"free": "form", "data": "example"};
+
+    final ExperienceEvent experienceevent = ExperienceEvent({
+      "xdmData": xdmData,
+      "data": data,
+      "datasetIdentifier": datasetId,
+    });
+    try {
+      result = await Edge.sendEvent(experienceevent);
```

Comment:
Can you paste the event sent by this API call, so we can verify the implementationDetails like version attached in the request to konductor is correct.

Reply:
{
	"meta": {
		"konductorConfig": {
			"streaming": {
				"enabled": true,
				"recordSeparator": "\u0000",
				"lineFeed": "\n"
			}
		},
		"state": {
			"entries": [{
				"key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_cluster",
				"value": "or2",
				"maxAge": 1800
			}, {
				"key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_identity",
				"value": "CiYzMTYyOTc0MzgzNTEwMTQzNTI3NjUzNzQwNDc1ODAyNjkzOTE3MlIOCJn9lo-lMBgBKgNPUjLwAZn9lo-lMA==",
				"maxAge": 34128000
			}]
		}
	},
	"xdm": {
		"identityMap": {
			"ECID": [{
				"primary": false,
				"id": "31629743835101435276537404758026939172",
				"authenticatedState": "ambiguous"
			}]
		},
		"implementationDetails": {
			"version": "3.4.2+1.3.0",
			"name": "https:\/\/ns.adobe.com\/experience\/mobilesdk\/ios\/flutter",
			"environment": "app"
		}
	},
	"events": [{
		"xdm": {
			"_id": "92464BE0-1DC6-4DE5-913D-02B9814B700A",
			"eventType": "SampleEventType",
			"timestamp": "2022-07-31T01:48:02Z"
		},
		"data": {
			"free": "form",
			"data": "example"
		}
	}]
}

Reply:
Android:


 {
      "xdm": {
        "implementationDetails": {
          "environment": "app",
          "name": "https:\/\/ns.adobe.com\/experience\/mobilesdk\/android\/flutter",
          "version": "1.11.1+1.3.2"
        },
        "identityMap": {
          "ECID": [
            {
              "id": "12749448659673225666642179032797304370",
              "authenticatedState": "ambiguous",
              "primary": false
            }
          ]
        }
      },
      "meta": {
        "state": {
          "entries": [
            {
              "maxAge": 1800,
              "value": "va6",
              "key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_cluster"
            },
            {
              "maxAge": 34128000,
              "value": "CiYxMjc0OTQ0ODY1OTY3MzIyNTY2NjY0MjE3OTAzMjc5NzMwNDM3MFIOCK3jnqCgMBgBKgNPUjLwAdbngtulMA==",
              "key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_identity"
            }
          ]
        },
        "konductorConfig": {
          "streaming": {
            "lineFeed": "\n",
            "enabled": true,
            "recordSeparator": "\u0000"
          }
        }
      },
      "events": [
        {
          "xdm": {
            "eventType": "SampleEventType",
            "_id": "676741e3-873d-46d3-9fa2-38804f621ee7",
            "timestamp": "2022-08-01T21:48:01.745Z"
          },
          "data": {
            "data": "example",
            "free": "form"
          }
        }
      ]
    }